runtime/src/bank/builtins/mod.rs

```diff
@@ -74,6 +118,194 @@ pub static BUILTINS: &[BuiltinPrototype] = &[
 ];
 
 pub static STATELESS_BUILTINS: &[StatelessBuiltinPrototype] = &[StatelessBuiltinPrototype {
+    #[cfg(not(test))]
+    core_bpf_migration_config: None,
+    #[cfg(test)]
+    core_bpf_migration_config: Some(test_only::feature_gate_program::CONFIG),
     program_id: solana_sdk::feature::id(),
     name: "feature_gate_program",
 }];
+
+#[cfg(test)]
+mod test_only {
+    use super::core_bpf_migration::{CoreBpfMigrationConfig, CoreBpfMigrationTargetType};
+    pub mod system_program {
+        pub mod feature {
+            solana_sdk::declare_id!("AnjsdWg7LXFbjDdy78wncCJs9PyTdWpKkFmHAwQU1mQ6");
+        }
+        pub mod source_program {
+            solana_sdk::declare_id!("EDEhzg1Jk79Wrk4mwpRa7txjgRxcE6igXwd6egFDVhuz");
+        }
```

Comment:
Where are all these ids from? Keypairs you created? If they are arbitrary, fine, but otherwise seems brittle.

Reply:
Yeah just arbitrary addresses because they have to be static.

Reply:
Sure!